server/db/driver/pg/internal/matches.go

```diff
@@ -79,6 +79,19 @@ const (
 	UpsertMatch = InsertMatch + ` ON CONFLICT (matchid) DO
 	UPDATE SET quantity = $10, status = $12;`
 
+	InsertCancelMatch = `INSERT INTO %s (matchid, active,
+			takerOrder, takerAccount, -- no taker address for a cancel order
+			makerOrder, makerAccount, -- omit maker's swap address too
+			epochIdx, epochDur,
+			quantity, rate, status)
+		VALUES ($1, FALSE, -- no active swap for a cancel
```

Comment:
well.... perhaps these matches should only be flagged inactive and status complete after SaveMatchAckSigA and SaveMatchAckSigB.  Hmm.

Reply:
Actually, I think it might be a bug that Swapper.Negotiate requests match acks for such cancel matches!

Reply:
What to do depends on whether it's important for the user to sign the match ack.  I don't see a penalization and there's not really any purpose for a cancel match.  @buck54321 thoughts?
But this PR fixes the main problem that was the server returning cancel order matches as active to the client.

Reply:
> Actually, I think it might be a bug that Swapper.Negotiate requests match acks for such cancel matches (at least from both parties since they are the same person).

I wouldn't call it a bug, but it's definitely inefficient to gather matches for both maker and taker there.

> What to do depends on whether it's important for the user to sign the match ack. 

I think it is important in general, but if you mean for the taker on a cancel order specifically, that could be dropped I think.  We don't really have a coherent penalization system yet.

Reply:
I'm just trying to think if it ever makes sense to consider a cancel order match as active so that it is provided to the user on connect.

The main purpose of the matches in the connect message are to facilitate trade/swap resumption, and the client can't do much with cancel matches in the connect message (only `handleMatchRoute` can ack to the message since it has the `msgjson.Message`, although `authDEX>parseMatches` can generate the `Acknowledgement`).

But if the user's client placed a cancel order and then shutdown before matching cycle and/or the match ack request was received, then on dexc startup the client would want to acknowledge the match.  So how could a client in this situation ever provide the ack that the server needed?

Resolutions: (1) Always consider a cancel order match inactive for the purposes of the connect message, and do not bother (re)soliciting a cancel match ack for reconnecting clients, or (2) include cancel matches lacking ack sigs as active, and have the server (re)send missing match ack requests on client connect.

This PR just considers all cancel orders as inactive immediately after they are made, not waiting for match acks to flag the match as inactive.  That does not prevent clients from acking and the sigs from being stored.

Reply:
This all makes sense. This PR is a good starting point, but maybe something more like resolution 2) should ultimately be the goal. Might be worth a separate issue.